folio-spring-system-user/src/main/java/org/folio/spring/context/ExecutionContextBuilder.java

```diff
@@ -22,38 +21,15 @@ public class ExecutionContextBuilder {
   private final FolioEnvironment folioEnvironment;
   private final FolioModuleMetadata moduleMetadata;
 
-  public FolioExecutionContext forSystemUser(SystemUser systemUser) {
-    var okapiUrl = systemUser.okapiUrl();
-    var tenantId = systemUser.tenantId();
-    var token = systemUser.token() == null ? null : systemUser.token().accessToken();
-    var userId = systemUser.userId();
-
-    return buildContext(okapiUrl, tenantId, token, userId, null);
+  public FolioExecutionContext forSystemUser(SystemUser systemUser, @CheckForNull Supplier<SystemUser> refresher) {
+    return new SystemUserExecutionContext(moduleMetadata, systemUser, refresher);
   }
 
   public FolioExecutionContext buildContext(String tenantId) {
     var okapiUrl = folioEnvironment.getOkapiUrl();
-    return buildContext(okapiUrl, tenantId, null, null, null);
-  }
-
-  private FolioExecutionContext buildContext(String okapiUrl, String tenantId, String token, String userId,
-                                             String requestId) {
-    Map<String, Collection<String>> headers = new HashMap<>();
-    if (isNotBlank(okapiUrl)) {
-      headers.put(XOkapiHeaders.URL, singleton(okapiUrl));
-    }
-    if (isNotBlank(tenantId)) {
-      headers.put(XOkapiHeaders.TENANT, singleton(tenantId));
-    }
-    if (isNotBlank(token)) {
-      headers.put(XOkapiHeaders.TOKEN, singleton(token));
-    }
-    if (isNotBlank(userId)) {
-      headers.put(XOkapiHeaders.USER_ID, singleton(userId));
-    }
-    if (isNotBlank(requestId)) {
-      headers.put(XOkapiHeaders.REQUEST_ID, singleton(requestId));
-    }
-    return new DefaultFolioExecutionContext(moduleMetadata, headers);
+    return new DefaultFolioExecutionContext(
+      moduleMetadata,
+      Map.of(XOkapiHeaders.URL, singleton(okapiUrl), XOkapiHeaders.TENANT, singleton(tenantId))
+    );
```

Comment:
This logic moved to the `SystemUserFolioExecutionContext` (the default context is only used for pre-authed requests, when we're logging in, in which case the only header used is tenant)